mlflow/shap.py

```diff
@@ -574,7 +575,6 @@ def _load_explainer(explainer_file, model=None):
     import shap
 
     def inject_model_loader(in_file):
-        pickle.load(in_file)  # No-Op to move file pointer forward
```

Comment:
This seems like a breaking change, right? Can we still load the explainer logged by `mlflow <= 1.14.0`?

Reply:
No, that wouldn't be possible. Even SHAP (v0.39.0) will not be able to load the explainers logged by SHAP (v0.38.0). There were changes to the underlying serialization process. Hence this consequently affects mlflow.

Reply:
@vivekchettiar  Are we at least emitting enough version metadata to detect this situation and inform users how they can load their older explainers? What are we doing to protect against another break moving forward?

Reply:
@akshaya-a thanks for bringing this up, part of update to SHAP serialization was to track the version of the serialization code that generated the explainers. When SHAP encounters a saved explainer from a older version, it'll try best-effort deserialization otherwise it'll inform the user about the incompatibility and print our the version that generated the file. Also, we're adding downstream integration tests in SHAP to check if there is a breaking change in the future.